src/test/java/uk/gov/companieshouse/confirmationstatementapi/eligibility/impl/CompanyStatusValidationTest.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.companieshouse.confirmationstatementapi.eligibility.impl;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.confirmationstatementapi.eligibility.EligibilityFailureReason;
+import uk.gov.companieshouse.confirmationstatementapi.exception.EligibilityException;
+
+import java.util.Collections;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class CompanyStatusValidationTest {
+
+    private static final String ALLOWED_VALUE = "AllowedStatus";
+    private static final Set<String> ALLOWED_LIST = Collections.singleton(ALLOWED_VALUE);
+
+    private CompanyStatusValidation companyStatusValidation;
+
+    @BeforeEach
+    void init() {
+        companyStatusValidation = new CompanyStatusValidation(ALLOWED_LIST);
+    }
+
+    @Test
+    void validateDoesNotThrowOnAllowedValue() throws EligibilityException {
+        CompanyProfileApi companyProfileApi = new CompanyProfileApi();
+        companyProfileApi.setCompanyStatus(ALLOWED_VALUE);
+
+        companyStatusValidation.validate(companyProfileApi);
+    }
+
+    @Test
+    void validateThrowsOnDisallowedValue() throws EligibilityException {
```

Comment:
i'd also add a test to pass null into the rule,  nulls/undefineds can cause issues

Reply:
or rather, the company status set to null, maybe do both!